README.md

```diff
@@ -35,75 +40,88 @@ For a list of the windows features you can install, please visit this [technet a
 
 * windowsfeature makes use of Powershell so you will need to have at least version 2.0 installed in order to use this module.
 
-## Beginning
+## Usage
 
-To install a single windows feature such as dotnet 3.5:
+To install a single windows feature such as .NET 3.5:
 
-    windowsfeature { 'NET-Framework-Core':
-      ensure => present,
-    }
+```puppet
+windowsfeature { 'NET-Framework-Core':
+  ensure => present,
+}
+```
 
 To install several windows features as part of a large application such IIS:
 
+```puppet
+$iis_features = ['Web-Server','Web-WebServer','Web-Asp-Net45','Web-ISAPI-Ext','Web-ISAPI-Filter','NET-Framework-45-ASPNET','WAS-NET-Environment','Web-Http-Redirect','Web-Filtering','Web-Mgmt-Console','Web-Mgmt-Tools']
 
-    $iis_features = ['Web-Server','Web-WebServer','Web-Asp-Net45','Web-ISAPI-Ext','Web-ISAPI-Filter','NET-Framework-45-ASPNET','WAS-NET-Environment','Web-Http-Redirect','Web-Filtering','Web-Mgmt-Console','Web-Mgmt-Tools']
-
-    windowsfeature { $iis_features:
-      ensure => present,
-    }
+windowsfeature { $iis_features:
+  ensure => present,
+}
+```
 
 To install any associated management tools:
 
-    windowsfeature { 'Web-WebServer':
-      ensure => present,
-      installmanagementtools => true
-    }
+```puppet
+windowsfeature { 'Web-WebServer':
+  ensure                 => present,
+  installmanagementtools => true,
+}
+```
 
-To install all subfeatures:
+To install all subfeatures without having to list them all out:
 
-    windowsfeature { 'Web-WebServer':
-      ensure             => present,
-      installsubfeatures => true
-    }
+```puppet
+windowsfeature { 'Web-WebServer':
+  ensure             => present,
+  installsubfeatures => true,
+}
+```
 
 To install a feature and that requires a restart:
 
-    windowsfeature {'RDS-RD-Server':
-      ensure             => present,
+```puppet
+    windowsfeature { 'RDS-RD-Server':
+      ensure  => present,
       restart => 'true'
     }
+```
 
-## Usage
+## Reference
 
-### Classes and Defined Types
+### Types
```

Comment:
This should get linked to in the TOC also (sub of 'Reference').
Should eventually move the types docs into puppet strings docs, but that's still something that we're trying to sort out how to do while making it easy for users to view.

Reply:
I looked at the rabbitmq module and another module and couldn't find this being done.

Reply:
rabbitmq doesn't have these sections in the README at all, because they moved to puppet strings, so the documentation is inline with the types / classes they refer to.

But the TOC should ideally have the section headings when documented in README, e.g., 
https://github.com/puppetlabs/puppetlabs-stdlib/#reference (you'll notice this one even has sub-headings _within_ the reference section).